workspaces/adventure-pack/goodies/kotlin/src/traverse_level_order/Main.kt

```diff
@@ -0,0 +1,19 @@
+package traverse_level_order
+
+public data class TreeNode(val `val`: Int, var left: TreeNode? = null, var right: TreeNode? = null)
+
+public fun TreeNode?.traverseLevelOrder(): Sequence<Collection<TreeNode>> {
```

Comment:
Out of curiosity, any reason to prefer Collection over List for the return type?